src/messaging/ExchangeHolder.h

```diff
@@ -20,6 +20,10 @@
 #include <lib/support/IntrusiveList.h>
 #include <messaging/ExchangeContext.h>
 
+#ifndef CHIP_EXCHANGE_HOLDER_DETAIL_LOGGING
+#define CHIP_EXCHANGE_HOLDER_DETAIL_LOGGING 0
+#endif // CHIP_EXCHANGE_HOLDER_DETAIL_LOGGING
+
```

Comment:
Curious why we require this? Isn't ChipLogDetail(..) enabled across the stack when CHIP_DETAIL_LOGGING is set? These macros seem to be adding another layer to enabling/disabling detail logs on top of what ChipLogDetail brings.

Reply:
Maybe the name is poor, but basically this would be a verbose setting for exchangeholder. Seems to be for debugging and it makes detail logging very noisy (and uses a bit more flash and cpu processing) when enabled.

We should have called this `CHIP_EXCHANGE_HOLDER_HIGH_VERBOSITY_LOGGING` or something like that. But for now this just fixes -Wundef